### ISSUE TITLE
Index out of range panic in solr input

### ISSUE THREAD

**Issue by windspear** [influxdata/telegraf#3604]
CentOS release 6.4 
Telegraf v1.5.0 (git: release-1.5 a1668bbf)

 use solr plugin as default config

[[inputs.solr]]
  ## specify a list of one or more Solr servers
  servers = ["http://localhost:8983"]
  ##
  ## specify a list of one or more Solr cores (default - all)
  # cores = ["main"]

i got this error ,how to fix it ? thank you

Error in plugin [inputs.solr]: Get http://localhost:8983/solr/resource/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
panic: runtime error: index out of range

goroutine 475 [running]:
github.com/influxdata/telegraf/plugins/inputs/solr.addCoreMetricsToAcc(0x1d9cf80, 0xc4200dcdc0, 0xc4208a8fc8, 0x8, 0xc42073efc0, 0xbe86a97a80425d32, 0x1ca1ec2193, 0x1e89820, 0x4, 0xc4203fb660)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/solr/solr.go:249 +0x5e0
github.com/influxdata/telegraf/plugins/inputs/solr.(*Solr).gatherServerMetrics.func1(0xc4208a9720, 0xc4203f4ff0, 0xbe86a97a80425d32, 0x1ca1ec2193, 0x1e89820, 0xc4207b0841, 0x18, 0xc4208a8fc8, 0x8, 0x1d9cf80, ...)
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/solr/solr.go:200 +0x173
created by github.com/influxdata/telegraf/plugins/inputs/solr.(*Solr).gatherServerMetrics
        /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/solr/solr.go:196 +0x261

**Comment by ljagiello** [comment]
Hi @windspear,

Could you provide more info? What Solr version are you using, what's the result of:
```
curl -svo result.txt http://localhost:8983/solr/resource/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE
```

**Comment by windspear** [comment]
solr version is 4.9.1

i use command curl -svo result.txt http://localhost:8983/solr/resource/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE
and got this

curl -svo result.txt http://localhost:8983/solr/resource/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE
[1] 23709
[2] 23710
[3] 23711
[4] 23712
[5] 23713
[2]   Done                    wt=json
[3]   Done                    cat=CORE
 * About to connect() to localhost port 8983 (#0)
*   Trying ::1... connected
* Connected to localhost (::1) port 8983 (#0)
> GET /solr/resource/admin/mbeans?stats=true HTTP/1.1
> User-Agent: curl/7.19.7 (x86_64-redhat-linux-gnu) libcurl/7.19.7 NSS/3.13.6.0 zlib/1.2.3 libidn/1.18 libssh2/1.4.2
> Host: localhost:8983
> Accept: */*
> 
< HTTP/1.1 200 OK
< Cache-Control: no-cache, no-store
< Pragma: no-cache
< Expires: Sat, 01 Jan 2000 01:00:00 GMT
< Last-Modified: Thu, 21 Dec 2017 08:05:13 GMT
< ETag: "160781ad93a"
< Content-Type: application/xml; charset=UTF-8
< Transfer-Encoding: chunked
< 
{ [data not shown]
* Failed writing body (0 != 16125)
* Failed writing data
* Closing connection #0

and must put enter ,got this 
[1]   Exit 23                 curl -svo result.txt http://localhost:8983/solr/resource/admin/mbeans?stats=true
[4]-  Done                    cat=QUERYHANDLER
[5]+  Done                    cat=UPDATEHANDLER
[result.txt](https://github.com/influxdata/telegraf/files/1578390/result.txt)

**Comment by ljagiello** [comment]
Try with quotes:
```
curl -svo result.txt "http://localhost:8983/solr/resource/admin/mbeans?stats=true&wt=json&cat=CORE&cat=QUERYHANDLER&cat=UPDATEHANDLER&cat=CACHE"
```

**Comment by windspear** [comment]
that's ok

[result.txt](https://github.com/influxdata/telegraf/files/1581336/result.txt)

**Comment by danielnelson** [comment]
Fix will be in 1.5.2, thanks to @mkboudreau
